rasa/nlu/model.py

```diff
@@ -307,25 +313,63 @@ def load(
             model_dir: The path of the model to load
             component_builder: The
                 :class:`rasa.nlu.components.ComponentBuilder` to use.
+            new_config: Optional new config to use for the new epochs.
+            finetuning_epoch_fraction: Value to multiply all epochs by.
 
         Returns:
             An interpreter that uses the loaded model.
         """
-
         model_metadata = Metadata.load(model_dir)
 
+        if new_config:
+            Interpreter._update_epochs_from_new_config(
+                model_metadata, new_config, finetuning_epoch_fraction
+            )
+
         Interpreter.ensure_model_compatibility(model_metadata)
-        return Interpreter.create(model_metadata, component_builder, skip_validation)
+        return Interpreter.create(
+            model_metadata,
+            component_builder,
+            skip_validation,
+            should_finetune=new_config is not None,
+        )
+
+    @staticmethod
+    def _update_epochs_from_new_config(
+        model_metadata: Metadata,
+        new_config: Optional[Dict] = None,
+        finetuning_epoch_fraction: float = 1.0,
+    ):
+        for p1, p2 in zip(model_metadata.metadata["pipeline"], new_config["pipeline"]):
+            if not p1.get("name") == p2.get("name"):
+                raise InvalidModelError("Inconsistent config for model to fine-tune.")
+            if "epochs" in p1:
+                p1["epochs"] = ceil(
+                    p2.get("epochs", p1["epochs"]) * finetuning_epoch_fraction
```

Comment:
We always want the epochs mentioned in the `new_config` to be multiplied by the `epoch_fraction`. The way it's implemented currently would be problematic if `epochs` is not defined explicitly in the config.

Reply:
cc @wochinge Related to our discussion on the slack [thread](https://rasa-hq.slack.com/archives/C8LEHMTAM/p1607103644498300)

Reply:
What is the desired behaviour if epochs are not defined in the new config? With this implementation it would multiply the epochs in the previous config by `epoch_fraction`, is that not what we want?

Reply:
Since the component would have used the default number of epochs in that case, we should multiply the `epoch_fraction` with the default number of epochs IMO. That would make the behaviour consistent with the idea of `epoch_fraction` - 
> We will always multiply the epochs based on the provided config

Reply:
Would be great to have constant for `epochs`. This would also ensure that other components don't name the `epochs` param differently for some reason.

Reply:
> Since the component would have used the default number of epochs in that case, we should multiply the epoch_fraction with the default number of epochs IMO. That would make the behaviour consistent with the idea of epoch_fraction

So if the model is trained first with config1 which has 10 epochs, and then is fine-tuned with config2 that has no epochs defined and epoch_fraction=0.5, then we ignore the 10 from the old config and instead multiply the default number of epochs by 0.5?

Reply:
Yes. We basically consider the current config as the source of truth. @wochinge Just making sure, you also agree with this?

Reply:
it's a little harder to implement but it makes sense to me conceptually.

Reply:
I am getting circular dependencies if I try to use `rasa.nlu.registry.get_component_class` to get the default epochs for a component... Another alternative is to pass `finetuning_epoch_fraction` into each component using the `context`. However this will require supporting it in all the components.

Reply:
Makes sense. I'd expect that this is a very rare case though (e.g. the auto config feature suggests a config which includes a specified number of epochs).

Reply:
> Another alternative is to pass finetuning_epoch_fraction into each component using the context. However this will require supporting it in all the components.

Mhm, but `epochs` doesn't make sense for some components (e.g. the `CountVectorizer`). 

> I am getting circular dependencies if I try to use rasa.nlu.registry.get_component_class to get the default epochs for a component.

How about a local import?